vault/expiration.go

```diff
@@ -1018,6 +1024,7 @@ func (m *ExpirationManager) revokeCommon(ctx context.Context, leaseID string, fo
 	m.removeFromPending(ctx, leaseID, true)
 	m.nonexpiring.Delete(leaseID)
 	m.irrevocable.Delete(leaseID)
+	m.irrevocableLeaseCount--
```

Comment:
could this lead to under-counting? there are likely scenarios where we hit this code block, but the lease doesn't exist in the `irrevocable` map (the above call would just be a noop)

Reply:
ah this would definitely lead to undercounting!!

Reply:
df2931a